src/main/java/com/netgrif/application/engine/petrinet/service/PetriNetService.java

```diff
@@ -158,8 +158,17 @@ public ImportPetriNetEventOutcome importPetriNet(InputStream xmlFile, String rel
 
     @Override
     public ImportPetriNetEventOutcome importPetriNet(InputStream xmlFile, VersionType releaseType, LoggedUser author) throws IOException, MissingPetriNetMetaDataException, MissingIconKeyException {
-        Optional<PetriNet> imported = getImporter().importPetriNet(copy(xmlFile));
         ImportPetriNetEventOutcome outcome = new ImportPetriNetEventOutcome();
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        byte[] buffer = new byte[1024];
+        int len;
+        while ((len = xmlFile.read(buffer)) > -1) {
+            baos.write(buffer, 0, len);
+        }
+        baos.flush();
+        InputStream xmlCopy1 = new ByteArrayInputStream(baos.toByteArray());
+        InputStream xmlCopy2 = new ByteArrayInputStream(baos.toByteArray());
+        Optional<PetriNet> imported = getImporter().importPetriNet(xmlCopy1);
```

Comment:
Why is the imported process file 4 times in RAM (like xmlFile, baos, xmlCopy1, xmlCopy2)?